test/bond-factory.test.ts

```diff
@@ -44,6 +45,19 @@ describe('Bond Factory contract', () => {
         )
     })
 
+    describe('proxy upgrade', () => {
+        it('maintains version across upgrade', async () => {
```

Comment:
@CjHare Is this the unit test you were expecting?

Reply:
Nearly 😉 

Add an `expect` to verify that the `originalVersion` is not `mock_tag`, otherwise looks good 👍

Reply:
Done